lib/api/model/model.dart

```diff
@@ -698,3 +710,82 @@ class DmMessage extends Message {
   @override
   Map<String, dynamic> toJson() => _$DmMessageToJson(this);
 }
+
+enum MessageEditState {
+  none,
+  edited,
+  moved;
+
+  // Adapted from the shared code:
+  //   https://github.com/zulip/zulip/blob/1fac99733/web/shared/src/resolved_topic.ts
+  // The canonical resolved-topic prefix.
+  static const String _resolvedTopicPrefix = '✔ ';
+
+  static bool _isResolved(String topic) {
+    return topic.startsWith(_resolvedTopicPrefix);
+  }
+
+  /// Whether two topics are equal, ignoring any resolved-topic prefix.
```

Comment:
Rereading this summary line (and the new name I suggested at https://github.com/zulip/zulip-flutter/pull/750#discussion_r1657943540 to match it), and then the implementation, I realize that this isn't actually accurate!

If we'd implemented a method that met this description, then it would be reflexive — `sameTopicModuloResolve(topic, topic)` would be true for any `topic`. But it isn't; in fact that's always `false`.

This is actually good because it also helps resolve the concern that I was otherwise thinking of here (and was going to say should be a TODO comment): in general Zulip topics are case-insensitive, so if this really meant a version of "these topics are equal" or "same topic", then it should be case-insensitive too.

I think what this method actually is about is: does this topic move reflect a resolve or unresolve operation. That explains the lack of case insensitivity: a resolve or unresolve wouldn't have changed case. And it explains the irreflexivity: a resolve or unresolve always changes the topic.

Reply:
Hmm, this difference is quite subtle. We can rewrite this as "Whether the two topics differs and only differs by whether one of them is resolved or not."